Firmware/code.py

```diff
@@ -5,72 +5,66 @@
 from adafruit_hid.keyboard import Keyboard
 from adafruit_hid.keycode import Keycode
 
+
+def init_button(btn):
+    new_btn = digitalio.DigitalInOut(btn)
+    new_btn.direction = digitalio.Direction.INPUT
+    new_btn.pull = digitalio.Pull.DOWN
+    return new_btn
+
+
+def press_button(btn, release=True):
+    keyboard.press(btn)
+    if release:
+        time.sleep(0.1)
+        keyboard.release(btn)
+
+
 #Setting pin varables
 Screen_1_Btn_Pin = board.GP1
 Screen_2_Btn_Pin = board.GP2
 Screen_3_Btn_Pin = board.GP3
 Screen_4_Btn_Pin = board.GP4
 Censor_Btn_Pin = board.GP15
 
-
 #..
 keyboard = Keyboard(usb_hid.devices)
 
 #Screen Button 1
-Screen_1_Btn = digitalio.DigitalInOut(Screen_1_Btn_Pin)
-Screen_1_Btn.direction = digitalio.Direction.INPUT
-Screen_1_Btn.pull = digitalio.Pull.DOWN
+Screen_1_Btn = init_button(Screen_1_Btn_Pin)
 
 #Screen Button 2
-Screen_2_Btn = digitalio.DigitalInOut(Screen_2_Btn_Pin)
-Screen_2_Btn.direction = digitalio.Direction.INPUT
-Screen_2_Btn.pull = digitalio.Pull.DOWN
+Screen_2_Btn = init_button(Screen_2_Btn_Pin)
 
 #Screen Button 3
-Screen_3_Btn = digitalio.DigitalInOut(Screen_3_Btn_Pin)
-Screen_3_Btn.direction = digitalio.Direction.INPUT
-Screen_3_Btn.pull = digitalio.Pull.DOWN
+Screen_3_Btn = init_button(Screen_3_Btn_Pin)
 
 #Screen Button 4
-Screen_4_Btn = digitalio.DigitalInOut(Screen_4_Btn_Pin)
-Screen_4_Btn.direction = digitalio.Direction.INPUT
-Screen_4_Btn.pull = digitalio.Pull.DOWN
+Screen_4_Btn = init_button(Screen_4_Btn_Pin)
 
 #Censor Button
-Censor_Btn = digitalio.DigitalInOut(Censor_Btn_Pin)
-Censor_Btn.direction = digitalio.Direction.INPUT
-Censor_Btn.pull = digitalio.Pull.DOWN
+Censor_Btn = init_button(Censor_Btn_Pin)
 
 
 while True:
     if Censor_Btn.value:
         print("Censor Buttion pressed")
-        keyboard.press(Keycode.SPACEBAR)
-        time.sleep(0.1)
-        keyboard.release(Keycode.SPACEBAR)
+        press_button(Keycode.SPACEBAR, release=False)

```

Comment:
So you can't release the spacebar (yet)?

Reply:
No, we're holding it down as long as the button is pressed. Although I forgot to add an else for when the button isn't being pressed